python/ccxt/base/exchange.py

```diff
@@ -359,16 +348,16 @@ def fetch(self, url, method='GET', headers=None, body=None):
         except Timeout as e:
             self.raise_error(RequestTimeout, method, url, e)
 
-        except ConnectionError as e:
-            self.raise_error(ExchangeNotAvailable, url, method, e)
-
-        except (TooManyRedirects, RequestException) as e:
+        except TooManyRedirects as e:
             self.raise_error(ExchangeError, url, method, e)
 
         except HTTPError as e:
             self.handle_errors(response.status_code, response.reason, url, method, self.last_response_headers, self.last_http_response)
             self.handle_rest_errors(e, response.status_code, self.last_http_response, url, method)
-            self.raise_error(ExchangeError, url, method, e, self.last_http_response)
+            self.raise_error(ExchangeNotAvailable, url, method, e, self.last_http_response)
```

Comment:
Oh-oh, this may be a breaking change. We can't really make all errors recoverable (ExchangeNotAvailable) by default, as they aren't recoverable most of the time (ExchangeError)...

Reply:
The logic behind this was that a HTTPError usually means the exchange is down as oppose to the url being broken. Therefore I thought it would be suitable for it to inherit from NetworkError. 

Not much context is given about when it may be raised - http://docs.python-requests.org/en/master/api/#requests.HTTPError

Reply:
> Not much context is given about when it may be raised

From the naming, I guess, it means, a HTTP error code of >=400 ? Many exchanges actually report non-recoverable errors with codes above 400, and most of them still don't have any exchange-specific error handling, so this case of HTTP error is to account for that and to provide a default HTTP error code handler for a yet-unknown error, which is a surprise, and we usually want to stop if there's any unexpected surprise.

Reply:
By that logic what should the base class of all request errors, `RequestException`, raise? Presumably ExchangeError right? Althought from my testing it seems like for many exchanges ExchangeError is raised when really it should be a ExchangeNotAvailable, and the difference is really hard to distiguish. 

    livecoin {"success":false,"errorCode":429,"errorMessage":"Too Many Requests"} <class 'ccxt.base.errors.ExchangeError'> GUP/ETH  (should this be DDOSProtectionError?)

In my program I have implented a loop that will count the number of times an exchange doesn't respond and then will set a down attribute if it doesn't respond a certain number of times in a row. I haven't looked into different error types that much since they are not particularly accurate, at least in this point in time.